cEP-0016.md

```diff
@@ -0,0 +1,180 @@
+# Git Commit Content Inspection
+
+| Metadata |                                                   |
+| -------- | ------------------------------------------------- |
+| cEP      | 16                                                |
+| Version  | 1.0                                               |
+| Title    | Git commit content inspection                     |
+| Authors  | Kriti Rohillla <mailto:kritirohilla567@gmail.com> |
+| Status   | Implementation due                                |
+| Type     | Process                                           |
+
+## Abstract
+
+This document describes coala process rules which will be enforced by the
+GitCommitBear. Some new changes will be introduced in the implementation of
+GitCommitBear based on which the commits made to coala will be detected and
+verified.
+
+## Introduction
+
+The GitCommitBear at present performs a check on the content of regular commits
+made to coala. However, there are many special types of commit messages that
+should be used only in conjunction with patches containing a special type of
+content.
+
+For example, git revert commit which is used to revert any previous commit and
+git merge commit which is generated whenever someone performs a git rebase and
+tries to merge master onto the pull request. Such commits generate a default
+commit message that is very concise and not according to coala’s standards.
+The coala project wants to allow `git revert` commits, but does not want to
+accept `git merge` commits as the repositories are strictly fast-forward merges
+that do not have an extra commit.
+
+Also in cases where some tests are not required by any pull request made to
+coala, some special sequences can be added to git commit messages to instruct
+tools to operate in a certain way.
+
+This project is about detecting such commits and verifying that they are correct
+according to the configuration settings in .coafile
+
+## Proposed Change
+
+Here is the detailed implementation stepwise:
+
+1. We start by identifying different types of special commits namely revert,
+   merge and travis commits.
+2. Each type of commit is then handled as a separate case.
+3. For travis commits:
+   - Commit contents are checked to find `[ci skip]`.
```

Comment:
My comment at https://github.com/coala/cEPs/pull/114#discussion_r186923706 has been completely ignored.

Your cEP is *very* wrong in so many ways because of this 'ci skip' thing.  You need to properly understand what a a `ci skip` is.

Reply:
`[ci skip]` or `[skip ci]` works identically for both circle ci and travis ci. As much as I understand, GitCommitBear should check if a commit is made with `[ci skip]` or `[skip ci]` in its shortlog or body. For each such commit, it should check that files modified are not already present in `.coafile` or tests. For multiple commits pushed together, this check should be done separately on each commit patch.

I do realize that I need to change travis commits --> ci builds. And so function can be something like - `disables_ci_build()` instead of `is_travis_commit()`.  

Is there anything else that I am getting wrong? I will make these changes asap.

Reply:
Also [this](https://circleci.com/docs/1.0/skip-a-build/) says "Currently neither [ci skip] nor [skip ci] supports fork PRs."
Does this change the way I am planning to implement these changes?
@damngamerz @seblat @refeed @jayvdb